include/boost/geometry/algorithms/detail/partition.hpp

```diff
@@ -84,23 +92,34 @@ inline void divide_into_subsets(Box const& lower_box,
     }
 }
 
-template <typename ExpandPolicy, typename Box, typename InputCollection>
+template
+<
+    typename ExpandPolicy,
+    typename Box,
+    typename ForwardRange,
+    typename IteratorVector
+>
 inline void expand_with_elements(Box& total,
-                InputCollection const& collection,
-                index_vector_type const& input)
+                ForwardRange const& /*forward_range*/,
```

Comment:
I remember now why I decided to keep it: in order to allow for the possibility to pass it, if we want at a later stage, to any of the policies visitors and used in the algorithm.
If you think this is too much I can remove it.